homeassistant/components/growatt_server/sensor.py

```diff
@@ -0,0 +1,187 @@
+}
+
+SENSOR_TYPES = {**TOTAL_SENSOR_TYPES, **INVERTER_SENSOR_TYPES}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_PLANT_ID, default=DEFAULT_PLANT_ID): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Growatt sensor."""
+    username = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+    plant_id = config[CONF_PLANT_ID]
+    name = config[CONF_NAME]
+
+    api = growattServer.GrowattApi()
+
+    # Log in to api and fetch first plant if no plant id is defined.
+    login_response = api.login(username, password)
+    if not login_response["success"] and login_response["errCode"] == "102":
+        _LOGGER.error("Username or Password may be incorrect!")
+        return
+    user_id = login_response["userId"]
+    if plant_id == DEFAULT_PLANT_ID:
+        plant_info = api.plant_list(user_id)
+        plant_id = plant_info["data"][0]["plantId"]
+
+    # Get a list of inverters for specified plant to add sensors for.
+    inverters = api.inverter_list(plant_id)
+    entities = []
+    probe = GrowattData(api, plant_id, True)
+    for sensor in TOTAL_SENSOR_TYPES:
+        entities.append(
+            GrowattInverter(probe, f"{name} Total", sensor, f"{plant_id}-{sensor}")
+        )
+
+    # Add sensors for each inverter in the specified plant.
+    for inverter in inverters:
+        probe = GrowattData(api, inverter["deviceSn"], False)
+        for sensor in INVERTER_SENSOR_TYPES:
+            entities.append(
+                GrowattInverter(
+                    probe,
+                    f"{inverter['deviceAilas']}",
+                    sensor,
+                    f"{inverter['deviceSn']}-{sensor}",
+                )
+            )
+
+    add_entities(entities, True)
+
+
+class GrowattInverter(Entity):
+    """Representation of a Growatt Sensor."""
+
+    def __init__(self, probe, name, sensor, unique_id):
+        """Initialize a PVOutput sensor."""
+        self.sensor = sensor
+        self.probe = probe
+        self._name = name
+        self._state = None
+        self._unique_id = unique_id
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"{self._name} {SENSOR_TYPES[self.sensor][0]}"
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the sensor."""
+        return self._unique_id
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        return "mdi:solar-power"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self.probe.get_data(SENSOR_TYPES[self.sensor][2])
+
+    @property
+    def device_class(self):
+        """Return the device class of the sensor."""
+        return "power"
```

Comment:
All the entities can't return power as device class now with the different sensor types existing. Maybe add the device class to the sensor type dict?

Reply:
I will do that, should i return `""` or `None` when there is no good device class?